### ISSUE TITLE
Jest expo targets global instead of globalThis

### ISSUE THREAD

**Issue by timhooker** [expo/expo#21509]
### Minimal reproducible example

module.exports = {   preset: 'jest-expo', }

### Summary

# Updates to Jest as of 28

Jest is now targeting globalThis.
https://jestjs.io/blog/2022/04/25/jest-28#using-globalthis

When I bumped Jest to v29 as part of the latest react native upgrade, I found my tests were broken. Jest expo's mocks weren't available in my tests. I noticed that in the setup for the jest preset that jest-expo is targeting `global`. 

https://github.com/expo/expo/blob/main/packages/jest-expo/src/preset/setup.js#L13-L16

After copying jest-expo's setup file into my own env and changing out references to global with globalThis, I was able to fix my tests.

Should jest-expo be targeting globalThis?

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.2.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.14.0 - ~/.asdf/installs/nodejs/18/bin/node
      npm: 9.3.1 - ~/.asdf/plugins/nodejs/shims/npm
      Watchman: 2023.01.30.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - ~/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
      Android SDK:
        API Levels: 26, 28, 29, 30, 31, 32, 33
        Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0
        System Images: android-21 | Google APIs ARM 64 v8a, android-24 | Google APIs ARM 64 v8a, android-26 | Google APIs ARM 64 v8a, android-28 | Google ARM64-V8a Play ARM 64 v8a, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs ARM 64 v8a, android-30 | Google Play ARM 64 v8a, android-32 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2022.1 AI-221.6008.13.2211.9619390
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^48.0.0 => 48.0.4 
      react: 18.2.0 => 18.2.0 
      react-native: ^0.71.0 => 0.71.3 
    Expo Workflow: bare
    
```

**Comment by patrickomeara** [comment]
Thanks @Kudo for the fix, and thanks @timhooker for filing the bug report. This has been a blocker for me upgrading to 48. 👍🏻

**Comment by Kudo** [comment]
we've published `jest-expo@48.0.2` with the fix. please try to upgrade the version and see whether it works for you. thanks for reporting this issue with a great description and even solution 🔥

**Comment by timhooker** [comment]
@Kudo Thank you! I really appreciate how quickly you jumped on this. I've updated jest-expo and swapped out the preset in my tests. My tests are now in the green.
